File: Keepr/Services/VaultsService.cs
using System;
using Keepr.Models;
using Keepr.Repositories;

namespace Keepr.Services
{
    public class VaultsService
    {
        private readonly VaultsRepository _vaultsRepository;

        public VaultsService(VaultsRepository vaultsRepository)
        {
            _vaultsRepository = vaultsRepository;
        }


        public Vault CreateVault(Vault data)
        {
            return _vaultsRepository.CreateVault(data);
        }

        public Vault GetById(int id, string userId)
        {
            Vault foundVault = _vaultsRepository.GetById(id);
            if(foundVault == null)
            {
                throw new Exception("Cannot find a vault by that Id");
            }
            if (foundVault.IsPrivate == true && foundVault.CreatorId != userId)
            {
                throw new Exception("That vault is private, sorry!");
            }
            return foundVault;
        }

        public Vault Edit(Vault editedVault)
        {
            Vault foundVault = GetById(editedVault.Id, editedVault.CreatorId);
            if(foundVault.CreatorId != editedVault.CreatorId)
            {
                throw new Exception("Unauthorized");
            }
            foundVault.Name = editedVault.Name ?? foundVault.Name;
            foundVault.Description = editedVault.Description ?? foundVault.Description;
            foundVault.IsPrivate = editedVault.IsPrivate ?? foundVault.IsPrivate;
            return _vaultsRepository.Edit(foundVault);
        }

        public void Delete(string userId, int vaultId)
        {
            Vault foundVault = GetById(vaultId, userId);
            if(foundVault.CreatorId != userId)
            {
                throw new Exception("Unauthorized");
            }
            _vaultsRepository.Delete(vaultId);
        }


    }
}

File: Keepr.client/src/services/VaultsService.js
import { AppState } from "../AppState"
import { Keep } from "../models/Keep"
import { Vault } from "../models/Vault"
import { VaultKeep } from "../models/VaultKeep"
import { router } from "../router"
import { logger } from "../utils/Logger"
import { api } from "./AxiosService"

class VaultsService{

    async getVaultByVaultId(vaultId){
        AppState.currentVault = null
        const res = await api.get(`api/vaults/${vaultId}`)
        logger.log("current Vault", res.data)
        AppState.currentVault = res.data

    }

    async getKeepsByVaultId(vaultId){
        AppState.vaultKeeps = []
        const res = await api.get(`api/vaults/${vaultId}/keeps`)
        AppState.vaultKeeps = res.data.map(k => new VaultKeep(k))
        logger.log("here are all the keeps in this vault", AppState.vaultKeeps)
        vaultsService.checkIfYourVault(vaultId)
    
    }
    async checkIfYourVault(vaultId){
        debugger
        const foundVault = await api.get(`api/vaults/${vaultId}`)
        if(foundVault.isPrivate == true && foundVault.creatorId !== AppState.account.id){
        router.push({ name: 'Home'})
        }
    
    }

    async createVault(vaultData) {
        const res = await api.post('api/vaults', vaultData)
        AppState.profileVaults.push(new Vault(res.data))
    }
    async createVaultKeep(keepId, vaultId) {
        var vaultKeepData = {}
        vaultKeepData.keepId = keepId
        vaultKeepData.vaultId = vaultId
        const res = await api.post('api/vaultkeeps', vaultKeepData)
    }

    async deleteVault(vaultId) {
        const res = await api.delete(`api/vaults/${vaultId}`)
        AppState.profileVaults = AppState.profileVaults.filter(v => v.id !== vaultId)
    }
    async setBlankVault() {
        AppState.currentVault = null
        logger.log("Current Vault Appstate", AppState.currentVault)
    }

    

}
export const vaultsService = new VaultsService()